### ISSUE TITLE
Duplication with ender portal

### ISSUE THREAD

**Issue by GedeonGrays** [CrucibleMC/Crucible#46]
_Hello o/_

_A fairly old duplication that works on almost any server core._

**How duplication works:**

Everything is extremely simple. First of all, we send a friend to the End to wait for a “gift”, which should appear on the obsidian platform that is generated when you go to the End through the portal. Then we from the Overworld send any falling block (for example, sand, gravel or anvil) through the portal. It turns out that the block in the End drops out as an item, but also placed on an obsidian platform. So why do we need a friend? He must break the block that appeared on the platform, otherwise you will not receive two items.
It is important to note that you cannot duplicate alone, since when teleporting to the End, if no one is there, the block will not drop out as an item, but only placed on the platform. Therefore, you do not get double the amount of block. Also, if a friend does not break the block that appears on the platform, and you teleport after the sent block, then it will simply be replaced by air, because when teleporting through the portal, the platform will be regenerated (again).

**Demonstration of duplication:**

Oh, and we specially recorded a video for you.
**Link:** https://youtu.be/K09Lk72CT0Y

**Deserves attention:**

Also, please note that in the video, starting from [0:48](https://youtu.be/K09Lk72CT0Y?t=48) it is shown that the obsidian platform is restored every time when teleporting to the End. This allows you to unlimitedly duplicate obsidian, and remove any blocks in the 5x5x3 area (length, width, height). I would suggest doing something with it.

**Comment by juanmuscaria** [comment]
From my memory paper has a fix for portal dupes, we can try to implement those later.
> Also, please note that in the video, starting from 0:48 it is shown that the obsidian platform is restored every time when teleporting to the End. This allows you to unlimitedly duplicate obsidian, and remove any blocks in the 5x5x3 area (length, width, height). I would suggest doing something with it.

It's a vanilla mechanic, really hard to abuse and there are better ways to get obsidian, like creating nether portals or just using mods. We can add an option to disable the spawn regeneration later for servers who have protection and want to build around that area without having everything reseted.

**Comment by GedeonGrays** [comment]
> It's a vanilla mechanic, really hard to abuse and there are better ways to get obsidian, like creating nether portals or just using mods. We can add an option to disable the spawn regeneration later for servers who have protection and want to build around that area without having everything reseted.

You are absolutely right, this is not such a critical problem. I just hoped that there might be some aesthetic and simple way to fix even such a small problem. The config is suitable as a solution if the Paper fix cannot be implemented.
But the main problem still remains the duplication of any falling blocks (and thanks to many mods, it is easy to make a falling block from any other).
